src/runtime/manifest.ts

```diff
@@ -50,42 +51,45 @@ export class ManifestError extends Error {
 // TODO(shans): Make sure that after refactor Storage objects have a lifecycle and can be directly used
 // deflated rather than requiring this stub.
 export class StorageStub {
-  type: Type;
-  id: string;
-  name: string;
-  storageKey: string;
-  storageProviderFactory: StorageProviderFactory;
-  referenceMode = false;
-  originalId: string;
-  description: string; // Fake to match StorageProviderBase;
-  /** Trust tags claimed by this data store. */
-  claims: ClaimIsTag[];
-
-  constructor(type: Type, id: string, name: string, storageKey: string, storageProviderFactory: StorageProviderFactory, originalId: string, claims: ClaimIsTag[]) {
-    this.type = type;
-    this.id = id;
-    this.name = name;
-    this.storageKey = storageKey;
-    this.storageProviderFactory = storageProviderFactory;
-    this.originalId = originalId;
-    this.claims = claims;
-  }
-
-  get version(): number {
-    return undefined; // Fake to match StorageProviderBase.
-  }
-
-  async inflate() {
-    const store = await this.storageProviderFactory.connect(this.id, this.type, this.storageKey);
+  constructor(public readonly type: Type, 
+              public readonly id: string,
+              public readonly name: string,
+              public readonly storageKey: string,
+              public readonly storageProviderFactory: StorageProviderFactory,
+              public readonly originalId: string,
+                /** Trust tags claimed by this data store. */
+              public readonly claims: ClaimIsTag[],
+              public readonly description: string,
+              public readonly version?: number,
+              public readonly source?: string,
+              public readonly referenceMode: boolean = false,
+              public readonly model?: {}[]) {}
+
+  async inflate(storageProviderFactory?: StorageProviderFactory) {
+    const factory = storageProviderFactory || this.storageProviderFactory;
+    const store = this.isStatic()
+        ? await factory.construct(this.id, this.type, this.storageKey)
+        : await factory.connect(this.id, this.type, this.storageKey);
     assert(store != null, 'inflating missing storageKey ' + this.storageKey);
     store.originalId = this.originalId;
+    store.referenceMode = this.referenceMode;
+    store.name = this.name;
+    store.source = this.source;
+    store.description = this.description;
+    if (this.isStatic()) {
+      await (store as VolatileStorageProvider).fromLiteral({version: this.version, model: this.model});
+    }
     return store;
   }
 
   toLiteral() {
     return undefined; // Fake to match StorageProviderBase;
   }
 
+  isStatic(): boolean {
```

Comment:
@shans i don't like this name, do you have any suggestions?

Reply:
isBackedByManifest maybe?

Reply:
thanks, done.